# IS 362 Week 5 Assignment

Your task in this week’s assignment is to answer three questions:
1. What is the northernmost airport in the United States?
2. What is the easternmost airport in the United States?
3. On February 12th, 2013, which New York area airport had the windiest weather?

First step is to import pandas so we can start to manipulate the data.

In [5]:
import pandas as pd

Next step is to read our data so we can start answering the assignment questions:

Question one requires us to find the most northern airport in the United States. A simple sort of the data looking for the highest latitude should yield our results:

In [6]:
airports = pd.read_csv ("C:/Users/johnj/IS 362 Spring 2018/Week 5 Assignment 1/airports.csv")
airports.sort_values(by=["lat"], ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt   tz dst              tzone  
417  149 -5.0   A                 \N  
230   44 -9.0   A  America/Anchorage  
110   41 -9.0   A  America/Anchorage  
708   35 -9.0   A  America/Anchorage  
152   96 -9.0   A  America/Anchorage

The code has executed but the information seems incorrect. Dillant Hopkins Airport (EEN) reports as furthest north but that is not the case. EEN is located it seems in Vermont. When looking up the airport's coordinates it seems the latitue and longitude have been reversed in the airport.csv. When considering this error it seems that Wiley Airport is the most northern airport based out of Alaska.

Question number two requires us to find the most eastern airport, so once again we can sort the airport.csv using the same code but by longitude this time and get our results.

In [7]:
airports.sort_values(by=["lon"], ascending=False).head()

faa                           name        lat         lon   alt   tz  \
1290  SYA                   Eareckson As  52.712275  174.113620    98 -9.0   
942   MYF               Montgomery Field  32.475900  117.759000    17  8.0   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478  8.0   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149 -5.0   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45 -5.0   

     dst              tzone  
1290   A  America/Anchorage  
942    A     Asia/Chongqing  
396    A     Asia/Chongqing  
417    A                 \N  
444    A   America/New_York

In this example our data is incorrect again. Eareckson is listed as our top result and is located in Alaska but that is not our most eastern state. It seems the negative sign is missing from the data set which is giving us the incorrect result. It seems the first four results are missing the sign. When taking that into consideration it seems that Eastport becomes the clear winner in the results.

Question 3 asks us to find which airport had the most windiest weather on a specific date. To accomplish this I will need to use a different csv which will be the weather.csv that was provided for us.

First step is to import the csv so we can begin to analyze the data. I will iclude a line of code that will sort the data by wind gusts and for the specified date.

In [8]:
weather = pd.read_csv("C:/Users/johnj/IS 362 Spring 2018/Week 5 Assignment 1/weather.csv")
weather[(weather["month"]==2) & (weather["day"]==12) & (weather["year"]==2013)].sort_values(by=["wind_gust"], ascending=False).head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1015     EWR  2013      2   12     8  39.02  26.96  61.63     260.0   
18433    LGA  2013      2   12     7  42.98  26.06  50.94     290.0   
1024     EWR  2013      2   12    17  44.06  26.06  48.87     270.0   
18444    LGA  2013      2   12    18  44.06  23.00  43.02     300.0   
9728     JFK  2013      2   12    13  42.08  24.08  48.56     280.0   

       wind_speed    wind_gust  precip  pressure  visib             time_hour  
1015   1048.36058  1206.432388     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18433    23.01560    26.485892     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1024     21.86482    25.161598     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18444    21.86482    25.161598     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
9728     20.71404    23.837303     0.0    1011.6   10.0  2013-02-12T13:00:00Z

Our first result brings up Newark Airopt with a wind gust os 1206.432388 which is way out of proportion to the rest of our results. This leads me to believe the data is incorrect once again and is skewing our results.

When doing a search of wind gust history for Newark airport I came across a site which states that wind gusts for that day topped out at 25 mph which gives us a more realistic number. Discarding the incorrect data puts LGA at the top of the results for question number 3.

# Conclusion

1. What is the northernmost airport in the United States?
    Wiley Post Will Rogers Memorial (BRW)
2. What is the easternmost airport in the United States?
    Eastport Municipal Airport (EPM)
3. On February 12th, 2013, which New York area airport had the windiest weather?
    LaGuardia Airport (LGA)